link/src/main/java/com/stripe/android/link/ui/paymentmethod/PaymentMethodBody.kt

```diff
@@ -63,9 +83,60 @@ internal fun PaymentMethodBody(
         factory = PaymentMethodViewModel.Factory(linkAccount, injector, loadFromArgs)
     )
 
+    val activityResultLauncher = rememberLauncherForActivityResult(
+        contract = FinancialConnectionsSheetForLinkContract(),
+        onResult = viewModel::onFinancialConnectionsAccountLinked
+    )
+
+    val clientSecret by viewModel.financialConnectionsSessionClientSecret.collectAsState()
+
+    clientSecret?.let { secret ->
+        LaunchedEffect(secret) {
+            activityResultLauncher.launch(
+                FinancialConnectionsSheetActivityArgs.ForLink(
+                    FinancialConnectionsSheet.Configuration(
+                        financialConnectionsSessionClientSecret = secret,
+                        publishableKey = viewModel.publishableKey
+                    )
+                )
+            )
+        }
+    }
+
     val formController by viewModel.formController.collectAsState()
 
-    if (formController == null) {
+    formController?.let {
```

Comment:
Nit: Can we give this a name, so that we avoid shadowing other `it`s down the line?